lib/files/checkFileAccess.ts

```diff
@@ -0,0 +1,49 @@
+import supabase from "@/lib/supabase/serverClient";
+
+/**
+ * Check if a user has access to a file
+ * 
+ * Access is granted if:
+ * - User is the file owner, OR
+ * - User has access to the artist that owns the file
+ * 
+ * @param userId - The user's account ID
+ * @param file - File record with owner_account_id and artist_account_id
+ * @returns true if user has access, false otherwise
+ */
+export async function checkFileAccess(
+  userId: string,
+  file: { owner_account_id: string; artist_account_id: string }
+): Promise<boolean> {
+  if (file.owner_account_id === userId) {
+    return true;
+  }
+
+  const { data: artistAccess } = await supabase
+    .from("account_artist_ids")
+    .select("artist_id")
+    .eq("account_id", userId)
+    .eq("artist_id", file.artist_account_id)
+    .maybeSingle();
```

Comment:
DRY / Single Responsibility Principle
- actual: there's already a lib to `select` on the `account_artist_ids` table based on `account_id` and `artist_id` yet this pull request ignores existing code and creates duplicate libs.
- required: use the existing lib. don't recreate the wheel.